### PR TITLE
Unittest for #1989

### PR DESCRIPTION
**Pull request by ramonsmits**

### CODE REVIEW

File: src/ServiceControl.Monitoring/Infrastructure/EndpointInstanceActivityTracker.cs [hunk 1]
```diff
@@ -17,14 +17,13 @@ public void Record(EndpointInstanceId instanceId, DateTime utcNow)
...
+            if (endpointsInstances.TryGetValue(endpointInstance, out var lastActivityTime))
             {
-                var timeSpan = DateTime.UtcNow - lastActivityTime;
-                return timeSpan > StalenessThreshold;
+                var age = DateTime.UtcNow - lastActivityTime;
+                return age > StalenessThreshold;
             }
 
-            return false;
+            return true;
```
Comment (WilliamBZA) [inline src/ServiceControl.Monitoring/Infrastructure/EndpointInstanceActivityTracker.cs:26]: This is a change in behaviour and causes `When_no_endpoint_instance_reports_not_stale` to fail. Is this change intentional?

File: src/ServiceControl.Monitoring.UnitTests/Infrastructure/EndpointInstanceActivityTrackerTests.cs [hunk 2]
```diff
@@ -17,7 +17,7 @@ public void Setup()
         [Test]
         public void When_no_endpoint_instance_reports_not_stale()
```
Comment (danielmarbach) [inline src/ServiceControl.Monitoring.UnitTests/Infrastructure/EndpointInstanceActivityTrackerTests.cs:18]: `When_no_endpoint_instance_reports_not_stale` to `When_no_endpoint_instance_reports_stale`

File: src/ServiceControl.Monitoring.UnitTests/API/AggregationTests.cs [hunk 3]
```diff
@@ -43,7 +44,52 @@ public void MetricAggregationByInstanceIsScopedToLogicalEndpoint()
...
+
+            var apiController = CreateController(processingTimeStore, endpointRegistry, out var activityTracker);
+
+            var endpointName = "Endpoint";
+
+            var instanceIds = new[] { "a", "b", "c" };
+            var instances = instanceIds.Select(instanceId => new EndpointInstanceId(endpointName, instanceId)).ToArray();
+
+            Array.ForEach(instances, instance => endpointRegistry.Record(instance));
+
```
Comment (danielmarbach) [inline src/ServiceControl.Monitoring.UnitTests/API/AggregationTests.cs:61]: 🔥

File: src/ServiceControl.Monitoring.UnitTests/API/AggregationTests.cs [hunk 4]
```diff
@@ -43,7 +44,52 @@ public void MetricAggregationByInstanceIsScopedToLogicalEndpoint()
...
         }
 
-        static DiagramApiController CreateConroller(ProcessingTimeStore processingTimeStore, EndpointRegistry endpointRegistry)
+        [Test]
+        public void ValidateIfConnectivityMathIsCorrect()
+        {
+            var processingTimeStore = new ProcessingTimeStore();
+            var endpointRegistry = new EndpointRegistry();
+
+            var apiController = CreateController(processingTimeStore, endpointRegistry, out var activityTracker);
```
Comment (danielmarbach) [inline src/ServiceControl.Monitoring.UnitTests/API/AggregationTests.cs:53]: Another option would be to have the activity tracker be a default parameter and then the static method creates it locally if not passed. This then allows to pass it in when needed without needing to do the crazy out trick
